doc: add adp536x documentation (#7607)
nrfconnect/sdk-nrf: doc/nrf/libraries/others/adp536x.rst

.. _lib_adp536x:

adp536x
#######

.. contents::
   :local:
   :depth: 2

The adp536x library provides an API to the ADP5360 :term:`power management integrated circuit (PMIC)`.

Implementation
==============

The library accesses the features of the ADP5360 PMIC by conducting Inter-Integrated Circuit (I2C) transactions with its registers.
Generic read, write, and masked write functions are private to the library.
The feature-specific functions are available to the user.

Supported features
==================

> The library provides several functions to set operating parameters for the ADP5360 PMIC and reading status values.

Reviewer: ```suggestion
The library provides several functions to set operating parameters for the ADP5360 PMIC and for reading status values.
```

.. _lib_adp536x:

adp536x
#######

.. contents::
   :local:
   :depth: 2

> The adp536x library provides an API to the ADP5360 power management IC.

Reviewer 2: ```suggestion
The adp536x library provides an API to the ADP5360 power management integrated circuit (PMIC). 
An I2C interface is required to communicate with the ADP5360 PMIC.
```
Moved the sentence from requirement to here as its a one liner.

Implementation
==============

> The library accesses the features of the ADP5360 by conducting I2C transactions to its registers. Generic read, write and masked write functions are private to the library, and feature-specific functions are exposed to the user.

Reviewer 2: ```suggestion
The library accesses the features of the ADP5360 PMIC by conducting I2C transactions with its registers. 
Generic read, write, and masked write functions are private to the library.
The feature-specific functions are available to the user.
```

Supported features
==================

> The library exposes a number of functions to set operating parameters of the ADP5360, as well as reading out status values.

Reviewer 2: ```suggestion
The library provides several functions to set operating parameters for the ADP5360 PMIC and reading status values.
```

Requirements
************

An I2C interface is required to communicate with the ADP5360.

> 

Reviewer 2: Please remove this section as its moved to line 11.

Configuration
*************

The library is enabled by the :kconfig:option:`CONFIG_ADP536X` configuration option. The I2C bus to use can be set with :kconfig:option:`CONFIG_ADP536X_BUS_NAME`.

Usage
*****

Initialize the library with :c:func:`adp536x_init`, and use the API to configure and read from the ADP5360 as desired.

> 

Reviewer 2: Could remove this section as its now in 'Configuration' section.

Dependencies
************

I2C-capable hardware

API documentation
*****************

> .. code-block::
> 
>    | Header file: :file:`include/adp536x`
>    | Source files: :file:`lib/adp536x/src`
> 
>    .. doxygengroup:: adp536x
>       :project: nrf
>       :members:

Reviewer 2: ```suggestion
| Header file: :file:`include/adp536x/adp536x.h`
| Source files: :file:`lib/adp536x.c`

.. doxygengroup:: adp536x
   :project: nrf
   :members:
```

No code block needed.
